endpoints/k8s/k8s

```diff
@@ -624,12 +644,19 @@ function build_pod_spec() {
         echo '          }' >>$json
         echo "        ]" >>$json
     fi
-    if [ "$type" == "cs" -a "$hugepage" == "1" ]; then
-        echo '       ,"volumeMounts": [' >>$json
-        echo '          { "mountPath": "/dev/hugepages",' >>$json
-        echo '            "name": "hugepage"' >>$json
-        echo '          }' >>$json
-        echo "        ]" >>$json
+    # If user provided volumeMounts.json, use it. It not, and "hugepage", do implicitly
+    if [ "$type" == "cs" ]; then
+        set +u
+        if [ ! -z "${volumeMounts[$name]}" ]; then
```

Comment:
`! -z` should be replaced with `-n`

Reply:
They are equivalent, but keeping it as-is will preserve the coding usage/style in this function. All of the other similar code blocks in this function use "! -z" construct and not "-n".